lib/ecl/smspec_node.cpp

```diff
@@ -439,73 +441,147 @@ char * smspec_alloc_local_completion_key( const char * join_string, const std::s
 }
 
 /*****************************************************************/
+static bool match_keyword_vector(int ipos, const std::vector<std::string> &vars, const std::string &keyword) {
+    //Validate string length    
+    if (!keyword[ipos]){
```

Comment:
This is incorrect. Use `keyword.size() < ipos` to check that the size of the string is at least `ipos` bytes long.

Reply:
The reason I did `!keyword[0]` to test for emptiness is that C strings are required to terminate with a null (`'\0'`), meaning that they're at least one byte long. Strings aren't allowed to contain non-terminating null terminators for obvious reason, so this is a simple check, but it doesn't scale.